source/setup/components/RemoteFileTree.js

```diff
@@ -231,7 +232,7 @@ class RemoteFileTree extends Component {
     render() {
         return (
             <Choose>
-                <When condition={this.props.rootDirectory && this.props.rootDirectory.directories.length > 0}>
+                <When condition={this.props.rootDirectory && this.props.isLoading}>
```

Comment:
I think this condition will be more effective and will fix all issues:

```suggestion
                <When condition={this.props.rootDirectory && this.props.rootDirectory.path}>
```

The `isLoading` props that are added can be removed now :) 
Thanks a lot for your help!

Reply:
But is this always the case @sallar? Wouldn't checking network/loading be more accurate as you then don't need to care about the contents?

Reply:
I agree with you @perry-mitchell